normalizer/app/geo_processor.py

```
@@ -35,131 +39,190 @@ def replacer(match):
+            lon, lat = result[0]["geometry"]["coordinates"]
+            return lat, lon
+        return None, None
+    except Exception:
+        return None, None
+
+def get_best_latlng(index, address, api_key, gsi_check=True, distance_threshold=200, priority="gsi", logger=None):
+    lat1, lon1 = get_gmap_latlng(address, api_key)
+    lat2, lon2 = get_gsi_latlng(address) if gsi_check else (None, None)
+
+    if lat1 is None and lat2 is None:
+        if logger: logger(f"警告: '{address}' の座標取得に失敗しました。")
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)